src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -313,10 +329,13 @@ export function SavedObjectProvider(Promise, Private, Notifier, confirmModalProm
      * @param {object} [options={}]
      * @property {boolean} [options.confirmOverwrite=false] - If true, attempts to create the source so it
      * can confirm an overwrite if a document with the id already exists.
+     * @property {boolean} [options.isTitleDuplicateConfirmed=false] - If true, save allowed with duplicate title
+     * @property {func} [options.onTitleDuplicate] - function called if duplicate title exists.
+     * When not provided, confirm modal will be displayed asking user to confirm or cancel save.
      * @return {Promise}
      * @resolved {String} - The id of the doc
      */
-    this.save = ({ confirmOverwrite } = {}) => {
+    this.save = ({ confirmOverwrite, isTitleDuplicateConfirmed, onTitleDuplicate } = {}) => {
```

Comment:
what do you think about passing in defaults for the two boleans explicitly here?  Could help avoid bugs in the future if any `=== false` checks.